src/robot/reporting/expandkeywordmatcher.py

```diff
@@ -0,0 +1,41 @@
+#  Copyright 2008-2015 Nokia Networks
+#  Copyright 2016-     Robot Framework Foundation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from robot.errors import DataError
+from robot.model import TagPatterns
+from robot.utils import MultiMatcher, is_list_like, py2to3
+
+
+@py2to3
+class ExpandKeywordMatcher(object):
+
+    def __init__(self, expand_args):
+        self.matched_ids = []
+        if not expand_args:
+            expand_args = []
+        elif not is_list_like(expand_args):
+            expand_args = [expand_args]
+        names = [n[5:] for n in expand_args if n[:5].lower() == 'name:']
+        tags  = [p[4:] for p in expand_args if p[:4].lower() == 'tag:']
+        self._namematcher = MultiMatcher(names) if len(names) > 0 else None
+        self._tagmatcher = MultiMatcher(tags) if len(tags) > 0 else None
```

Comment:
I doubt this performance optimization really affects anything. If `--expandkeywords` has been used, either `_namematcher` or `_tagmatcher` need to be used anyway. Always creating them would simplify the `check()` implementation. If this is left, `if len(names) > 0` should be replaced with `if names`.

Reply:
improved code